ckanext/switzerland/helpers/validators.py

```diff
@@ -326,6 +326,11 @@ def ogdch_validate_formfield_publisher(field, schema):
     "it": "Italian Name"}, "url": "Publisher URL"}'
     """
     def validator(key, data, errors, context):
+        # the value is already a dict
+        if isinstance(data.get(key), dict):
+            data[key] = json.dumps(data.get(key))
+            return  # exit early since this case is handled
+        # the key is missing
         if not data.get(key):
```

Comment:
I think you can remove this if statement and un-indent everything beneath it by one tab

Reply:
I would not do this, actually I would remove 
```
if isinstance(data.get(key), dict):
            data[key] = json.dumps(data.get(key))
            return
```
And really validate the missing value here. If we will remove `if not data.get(key)`
It could lead to overwriting valid data, like str value (from foaf:Organization).